File: WorldFeed/Server/src/WorldFeed.Common/Public/Parameters/StreamsClient/CreateTrackedStreamParameters.cs
namespace WorldFeed.Common.Public.Parameters.StreamsClient
{
    using WorldFeed.Common.Settings;

    public interface ICreateTrackedTweetStreamParameters : ICustomRequestParameters
    {
        /// <summary>
        /// Decide whether to use Extended or Compat mode
        /// </summary>
        TweetMode? TweetMode { get; set; }
    }

    public class CreateTrackedTweetStreamParameters : CustomRequestParameters, ICreateTrackedTweetStreamParameters
    {
        public TweetMode? TweetMode { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Public/Parameters/StreamsClient/CreateTrackedStreamParameters.ts
import {CustomRequestParameters, ICustomRequestParameters} from "../CustomRequestParameters";
import {TweetMode} from "../../Settings/TweetinviSettings";

export interface ICreateTrackedTweetStreamParameters extends ICustomRequestParameters {
  // Decide whether to use Extended or Compat mode
  tweetMode?: TweetMode;
}

export class CreateTrackedTweetStreamParameters extends CustomRequestParameters implements ICreateTrackedTweetStreamParameters {
  public tweetMode?: TweetMode;
}
